src/inmanta/server/services/compilerservice.py

```diff
@@ -652,10 +653,13 @@ async def request_recompile(
         """
         if in_db_transaction and not connection:
             raise Exception("A connection should be provided when in_db_transaction is True.")
+        # TODO: can we afford this "breaking" change?
+        #   -> breaks current stable inmanta-lsm (can be worked around here), works without
+        #   -> breaks old inmanta-lsm, compiles may hang without
         if in_db_transaction != (connection is not None and connection.is_in_transaction()):
             raise Exception(
                 f"in_db_transaction is {in_db_transaction},"
```

Comment:
I added this error in the original PR. It is technically a breaking change but I think we can afford it because we always release a full product anyway, and the only user is inmanta-lsm, and the previous behavior was bugged:
1. For current stable inmanta-lsm, this change would raise an exception because of a typing error in inmanta-lsm. If the exception weren't raised, the behavior would be as expected.
2. For older inmanta-lsm than current stable, this change would also raise an exception because it uses a transaction without setting the flag. If the exception weren't raised, the behavior would still be unstable (this is why we added the flag in the first place).

Let me know if you think we have to remain compatible.